hibernate-core/src/main/java/org/hibernate/dialect/function/DB2FormatEmulation.java

```diff
@@ -23,9 +24,10 @@
  *
  * @author Gavin King
  */
-public class DB2FormatEmulation extends AbstractSqmFunctionDescriptor {
```

Comment:
Minor I guess, but I had made the change for AbstractSqmSelfRenderingFunctionDescriptor to simply AbstractSqmFunctionDescriptor because implementing SqmFunctionDescriptor that is not self-rendering really has no "commonality".  But I don't hate your reversion enough to ask you to change it back

Reply:
Like I commented in the text of the pull request:

> If some things in this PR don't seem to make 100% sense, it's because the new code that uses stuff here is not included in this PR.